src/bio/bio_buffer.c

```diff
@@ -536,37 +536,51 @@ copy_one(struct bio_desc *biod, struct bio_iov *biov, void *data)
 	return -DER_REC2BIG;
 }
 
-static int
-iterate_biov(struct bio_desc *biod,
-	     int (*cb_fn)(struct bio_desc *, struct bio_iov *, void *data),
-	     void *data)
+static void
+copy_one_setup(struct bio_desc *biod, int idx, void *data)
 {
-	int i, j, rc = 0;
+	struct bio_copy_args *arg = data;
 
-	for (i = 0; i < biod->bd_sgl_cnt; i++) {
-		struct bio_sglist *bsgl = &biod->bd_sgls[i];
+	D_ASSERT(idx < arg->ca_sgl_cnt);
+	arg->ca_sgl_idx = idx;
+	arg->ca_iov_idx = 0;
+	arg->ca_iov_off = 0;
+	if (biod->bd_type == BIO_IOD_TYPE_FETCH)
+		arg->ca_sgls[idx].sg_nr_out = 0;
+}
 
-		if (data != NULL) {
-			if (cb_fn == copy_one) {
-				struct bio_copy_args *arg = data;
+static void
+map_one_setup(struct bio_desc *biod, int idx, void *data)
+{
+	struct bio_bulk_args *arg = data;
 
-				D_ASSERT(i < arg->ca_sgl_cnt);
-				arg->ca_sgl_idx = i;
-				arg->ca_iov_idx = 0;
-				arg->ca_iov_off = 0;
-				if (biod->bd_type == BIO_IOD_TYPE_FETCH)
-					arg->ca_sgls[i].sg_nr_out = 0;
-			} else if (cb_fn == bulk_map_one) {
-				struct bio_bulk_args *arg = data;
+	arg->ba_sgl_idx = idx;
+}
 
-				arg->ba_sgl_idx = i;
-			}
-		}
+static int
+iterate_biov(struct bio_desc *biod, int (*cb_fn)(struct bio_desc *, struct bio_iov *, void *data),
+	     void            *data)
+{
+	void (*prep_fn)(struct bio_desc *, int, void *) = NULL;
+	int rc                                          = 0;
+
+	if (data != NULL) {
+		if (cb_fn == copy_one)
+			prep_fn = copy_one_setup;
+		else if (cb_fn == bulk_map_one)
+			prep_fn = map_one_setup;
+	}
+
+	for (int i = 0; i < biod->bd_sgl_cnt; i++) {
```

Comment:
This is the new coding style that declares a variable here?

Reply:
it's not really new, the coding standard has always said use the smallest scope that you can for variables.  Including this wasn't entirely deliberate, I just happened to have re-read the document the day before writing this and wasn't sure if it would work or not.